src/Microsoft.AspNet.Server.Kestrel/KestrelServerInformation.cs

```diff
@@ -39,5 +39,34 @@ private static ICollection<string> GetAddresses(IConfiguration configuration)
 
             return addresses;
         }
+
+        private static int GetThreadCount(IConfiguration configuration)
+        {
+            var threadCountString = configuration["server.threadCount"];
+
+            int threadCount;
+            if (string.IsNullOrEmpty(threadCountString) || !int.TryParse(threadCountString, out threadCount))
```

Comment:
What if I move the check into the ThreadCount setter? That way it would ensure that the value is always valid?